Software/Development/SemesterProject/App/encoderManager.c

```diff
@@ -0,0 +1,195 @@
+
+    // Inputs are synchronized to SYSCLKOUT by default.
+    GpioCtrlRegs.GPAQSEL2.bit.GPIO20 = 0;  // Sync to SYSCLKOUT GPIO20 (EQEP1A)
+    GpioCtrlRegs.GPAQSEL2.bit.GPIO21 = 0;  // Sync to SYSCLKOUT GPIO21 (EQEP1B)
+    GpioCtrlRegs.GPAQSEL2.bit.GPIO23 = 0;  // Sync to SYSCLKOUT GPIO23 (EQEP1I)
+
+    // This specifies which of the possible GPIO pins will be eQEP1 functional pins.
+    GpioCtrlRegs.GPAMUX2.bit.GPIO20 = 1;   // Configure GPIO20 as EQEP1A
+    GpioCtrlRegs.GPAMUX2.bit.GPIO21 = 1;   // Configure GPIO21 as EQEP1B
+    GpioCtrlRegs.GPAMUX2.bit.GPIO23 = 1;   // Configure GPIO23 as EQEP1I
+
+    //EQep1Regs.QUPRD = 900000;         // Unit Timer for 100Hz at 80 MHz SYSCLKOUT
+
+    EQep1Regs.QDECCTL.bit.QSRC = 00;  // QEP quadrature count mode
+
+    //EQep control:
+    EQep1Regs.QEPCTL.bit.FREE_SOFT = 2; // Position counter unaffected by emulation suspension.
+    EQep1Regs.QEPCTL.bit.PCRM = 00;   // PCRM=00 mode - QPOSCNT reset on index event
+    EQep1Regs.QEPCTL.bit.SEI = 00;    // Strobe event initialization of position counter. 00->does nothing.
+    EQep1Regs.QEPCTL.bit.IEI = 00;    //Do nothing --> 10;    // Index event initialization of counter. 10->init on rising edge.
+    EQep1Regs.QEPCTL.bit.SWI = 0;     // Software init of position counter. 0->do nothing.
+    EQep1Regs.QEPCTL.bit.SEL = 1;     // Strobe event latch of pos counter. 1-> Clockwise, latch on rise, counterclockwise, latch on fall.
+    EQep1Regs.QEPCTL.bit.IEL = 01;    // Index event latch of pos counter. 01->latches on rising edge of index.
+    EQep1Regs.QEPCTL.bit.QPEN = 1;    // Quadrature pos counter enable/software reset. 1->eQEP pos counter enabled.
+
+    //QCLM REGISTER DETERMINES CONSTANT X vs T.
+    EQep1Regs.QEPCTL.bit.QCLM = 1;    // eQEP capture latch mode. 1->latch on unit time out
+    EQep1Regs.QEPCTL.bit.UTE = 1;     // Unit Timeout Enable. 1->Enabled.
+    EQep1Regs.QEPCTL.bit.WDE = 0;     // Watchdog enable. 0->disabled
+
+    EQep1Regs.QPOSMAX = ENCODER_STEPS - 1;         // Maximum possible counter value.
+    EQep1Regs.QPOSINIT = 0;           // Position value used to initialize when index event.
+
+    EQep1Regs.QCAPCTL.bit.CEN = 1;    // QEP Capture Enable
+    EQep1Regs.QCAPCTL.bit.CCPS = 6;   // 1/64 for CAP clock prescaler
+    EQep1Regs.QCAPCTL.bit.UPPS = 5;   // 1/32 for unit position prescaler
+
+    EDIS;
+
+    motorPosSpeedConstructor();
+}
+
+
+/*
+ * Encoder constructor.
+ * To be called as motorPosSpeedConstructor(&motorPosSpeedObject);
+ */
+void motorPosSpeedConstructor(void)
+{
+    motorPosSpeedObject.poles = 2;
+    motorPosSpeedObject.dir = 0;
+    motorPosSpeedObject.thetaMech = 0;
+    motorPosSpeedObject.thetaElec = 0;
+    motorPosSpeedObject.thetaElecOld = 0;
+    motorPosSpeedObject.speedTempCount = 0;
+    motorPosSpeedObject.freqElec = 0;
+    motorPosSpeedObject.freqMech = 0;
+    motorPosSpeedObject.rpmMech = 0;
+    poles_inverse = (float) (1.0 / motorPosSpeedObject.poles);
```

Comment:
poles_inverse and motorPosSpeedObject.poles seems redundant. I think the proper way of doing it would be to use a #define poles_inverse number.